test/unit/math/prim/mat/functor/map_rect_concurrent_threads_test.cpp

```diff
@@ -0,0 +1,94 @@
+// the tests here check that map_rect_concurrent works correct as such we
+// enforce that STAN_MPI is NOT defined while STAN_THREADS is turned on
+
+#ifdef STAN_MPI
+#undef STAN_MPI
+#endif
+
+#ifndef STAN_THREADS
+#define STAN_THREADS
+#endif
```

Comment:
I'm a little unsure if we should have tests that change these compile flags like this versus changing them from the Jenkins file and running under various conditions? Like I think it might be better to just have these tests only execute if STAN_THREADS is defined, but maybe this is a question for @syclik.

Reply:
This test is intended to tickle the scheduling problems. The test does not use ad and as such it is only to get the scheduling code „active“ in order to test it. I think it is ok to always test it, but I see your point.

Reply:
It would be more like the GPU tests which are completely ifdef'd out if STAN_OPENCL isn't defined and only compile and run if it is defined. We have to go through all of these configurations anyway on Jenkins, seems like we might continue to just use that to trigger the proper tests...

I just realized - I think we do need to take these out and rely on external control of these make variables because otherwise if you are running all tests, whenever this one starts getting compiled it will change these flags for all subsequent tests, I think.

Reply:
Are u sure about that change for others? I thought that each cpp test file is its own translational unit and becomes and independent exe file.

Reply:
Oh yeah, you're right, so it's just a style thing. I don't feel strongly but would think we would wanna do this test the way we've done most of the other tests that use compile flags that I'm aware of - #ifdef it out completely if STAN_THREADS isn't defined. But we can ask @syclik for a final judgement on that.